### PR TITLE
Update country code config

### LINKED ISSUE #390: Implement loginID phone number widget
Design for phone number widget is updated

- [x] Country name translation -> i18n-iso-countries
- [x] Phone number country code to region code mapping
  - [x] script for generate hard coded map => google libphonenumber (getRegionCodesForCountryCode)
  - [x] use hard coded map for getting country code with calling code
- [x] Create list with selection and ordering in only selected item
- [x] selected all -> checked, partially selected -> indeterminate, none selected -> unchecked
- [x] Implement search box (local filtering, only filter unselected)
- [x] Implement search logic and filtering
- [x] Study mapping between config and UI state
- [x] Handle change config mutation

### LINKED ISSUE #487: Update country calling code config shape
ref #390

### PR DESCRIPTION
**Pull request by axolotl0212**
fix #487 
fix #390 

- updated backend country code config shape
- updated frontend UI

# Screenshot

Screenshot 2020-10-21 at 11 34 46 AM

### CODE REVIEW

**Comment by axolotl0212** [comment]
rebased

File: pkg/lib/config/ui.go [hunk 1]
```diff
@@ -23,22 +23,42 @@ var _ = Schema.Add("UICountryCallingCodeConfig", `
 	"type": "object",
 	"additionalProperties": false,
 	"properties": {
-		"values": { "type": "array", "items": { "type": "string" } },
-		"default": { "type": "string" }
+		"allow_list": { "type": "array", "items": { "type": "string" }, "minItems": 1 },
```
Comment (louischan-oursky) [inline pkg/lib/config/ui.go:26]: allowlist is one word.

File: pkg/lib/config/ui.go [hunk 2]
```diff
@@ -23,22 +23,42 @@ var _ = Schema.Add("UICountryCallingCodeConfig", `
...
 func (c *UICountryCallingCodeConfig) SetDefaults() {
-	if c.Values == nil {
-		c.Values = phone.CountryCallingCodes
+	if c.AllowList == nil {
+		c.AllowList = phone.CountryCallingCodes
 	}
-	if c.Default == "" {
-		c.Default = c.Values[0]
+	if c.PinnedList == nil {
+		c.PinnedList = []string{c.AllowList[0]}
```
Comment (louischan-oursky) [inline pkg/lib/config/ui.go:42]: The default value should be nil. So no special handling is needed.

File: pkg/auth/handler/webapp/validation.go [hunk 3]
```diff
@@ -45,6 +45,6 @@ func (p *FormPrefiller) Prefill(form url.Values) {
 	}
 
 	if _, ok := form["x_calling_code"]; !ok {
-		form.Set("x_calling_code", p.UI.CountryCallingCode.Default)
+		form.Set("x_calling_code", p.UI.CountryCallingCode.PinnedList[0])
```
Comment (louischan-oursky) [inline pkg/auth/handler/webapp/validation.go:48]: It should be the first item of GetActiveCountryCodes

File: portal/src/graphql/adminapi/AddPhoneScreen.tsx [hunk 4]
```diff
@@ -52,17 +53,20 @@ const AddPhoneForm: React.FC<AddPhoneFormProps> = function AddPhoneForm(
 
   const countryCodeConfig = useMemo(() => {
     const countryCodeConfig = appConfig?.ui?.country_calling_code;
-    const values = countryCodeConfig?.values ?? [];
+    const allowList = countryCodeConfig?.allow_list ?? [];
+    const pinnedList = countryCodeConfig?.pinned_list ?? [];
+    const values = getActiveCountryCallingCode(pinnedList, allowList);
+    const defaultCallingCode = pinnedList[0];
```
Comment (louischan-oursky) [inline portal/src/graphql/adminapi/AddPhoneScreen.tsx:59]: default should be the first item of active list.

File: portal/src/graphql/portal/AuthenticationLoginIDSettings.tsx [hunk 5]
```diff
@@ -186,9 +190,26 @@ function constructStateFromAppConfig(
...
+    pinnedCallingCodes,
   };
 }
 
+function localValidate(
+  renderToString: (messageID: string) => string,
+  state: AuthenticationLoginIDSettingsState
+): string | undefined {
+  if (
+    state.pinnedCallingCodes.length === 0 ||
```
Comment (louischan-oursky) [inline portal/src/graphql/portal/AuthenticationLoginIDSettings.tsx:202]: pinned list can be empty.

**Comment by axolotl0212** [comment]
@louischan-oursky updated

**Pull request merged**